ethereum/chain.py

```diff
@@ -1,4 +1,5 @@
 import time
+from itertools import count, islice
```

Comment:
*nitpick*

I'm a fan of just `import itertools` and then getting at the functions by referencing them from `itertools.count()`, etc.

This has the benefit of reducing cognitive overhead when parsing the code since you can see that it's `itertools.count` and not some custom `count` function.